src/IonEventStream.ts

```diff
@@ -206,17 +230,19 @@ export class IonEventStream {
             if(actualEvent.eventType == IonEventType.STREAM_END && expectedEvent.eventType == IonEventType.STREAM_END) {
                 break;
             } else if(actualEvent.eventType == IonEventType.STREAM_END || expectedEvent.eventType == IonEventType.STREAM_END) {
-                    throw new Error("Different number of comparison sets.");
+                throw new Error("Different number of comparison sets.");
             } else if (!(actualEvent.ionType == IonTypes.LIST || actualEvent.ionType == IonTypes.SEXP)
                 || !(expectedEvent.ionType == IonTypes.LIST || expectedEvent.ionType == IonTypes.SEXP)) {
                 throw new Error("Comparison sets must be lists or s-expressions.");
-            } else if((actualEvent.annotations[0] === 'embedded_documents') as any ^ (expectedEvent.annotations[0] === 'embedded_documents') as any) {
+            } else if(this.isEmbedded(actualEvent) as any ^ this.isEmbedded(expectedEvent) as any) {
```

Comment:
Why does the boolean expression `this.isEmbedded(...)` need to be cast `as any`?

Reply:
^ operator can only be applied to the number type, according to the TypeScript specification.